### PR TITLE
chore: add allocation info for cluster ui [DET-10018]

### PR DESCRIPTION
**Pull request by jgongd**
## Description

Add allocation id, task id, and job id to GetAgents endpoint.

## Test Plan

Integration tests

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

## Ticket

DET-10018

### CODE REVIEW

File: master/pkg/model/agent.go [hunk 1]
```diff
@@ -80,3 +81,32 @@ type AgentStats struct {
...
+
+// ToProto converts a ContainerSummary to its protobuf representation.
+func (c *ContainerSummary) ToProto() *containerv1.Container {
+	if c == nil {
+		return nil
+	}
+	var devices []*devicev1.Device
+	for _, d := range c.Devices {
+		devices = append(devices, d.Proto())
+	}
```
Comment (maxrussell) [inline master/pkg/model/agent.go:103]: It's totally fine if you wanna leave this how it is, but I like to define a new type that's a slice and define a `Proto()` method on that slice type, [like with `Groups`](https://github.com/determined-ai/determined/blob/3511abf5cc2a/master/pkg/model/user_group.go#L27).

File: master/internal/rm/agentrm/agent_state.go [hunk 2]
```diff
@@ -372,21 +375,37 @@ func (a *agentState) getSlotsSummary(baseAddress string) model.SlotsSummary {
...
+func (a *agentState) getContainerSummary(s *slot) *model.ContainerSummary {
+	cid := s.containerID
+	if cid == nil {
+		return nil
+	}
+
+	container := a.containerState[*cid]
+
+	fmt.Printf("container: %+v\n", container)
+	fmt.Printf("container info 2: %+v\n", a.containerAllocation[*cid])
```
Comment (maxrussell) [inline master/internal/rm/agentrm/agent_state.go:398]: Probably don't want these here.

**Review by maxrussell — CHANGES_REQUESTED** [review]
Other than the prints and the nit, LGTM.

File: master/internal/rm/agentrm/agent_state.go (hunk 2 above)
Comment (jgongd) [reply master/internal/rm/agentrm/agent_state.go:398]: Good catch

File: master/pkg/model/agent.go (hunk 1 above)
Comment (jgongd) [reply master/pkg/model/agent.go:103]: Neat

Comment (jgongd) [reply master/pkg/model/agent.go]: Is this for improving code quality, so all device slices has a Proto() instead of repeating the loops everywhere?

Comment (maxrussell) [reply master/pkg/model/agent.go]: Yeah, that exactly my reason.

**Review by maxrussell — APPROVED** [review]
Super tiny nit, but LGTM!

**Pull request merged**